datafusion/src/physical_plan/hash_join.rs

```diff
@@ -278,6 +279,14 @@ impl ExecutionPlan for HashJoinExec {
         self.right.output_partitioning()
     }
 
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
+        None
```

Comment:
Probably doesn't make sense to address this now, but order of the right side might be (fully or partially) maintained for hash joins.

Reply:
To give a concrete example, if the right side of the join is sorted on field `x` and we use an inner join, the output is sorted on `x` too as rows are not reordered.

Reply:
👍  makes sense.

I think that only applies for inner joins though (as outer joins may stick nulls into inconvenient places 😆 )

Reply:
Yes 😂 I think right join is the other one that maintains sortedness too.